### ISSUE TITLE
Improve alias generator

### ISSUE THREAD

**Issue by mayukh18** [mayukh18/BlindChat#1]
alias generator is the simplest right now. Needs to be made better.

**Comment by Arya04** [comment]
the function needs to be made better or just more nouns and adjectives added to the list?

**Comment by mayukh18** [comment]
Ideally the function is to be improved but additions to the existing lists are also welcome.

**Comment by gkeller2** [comment]
"made better" is not much of a guideline. What's the problem/limitation that you see in it?

**Comment by mayukh18** [comment]
Yes "made better" is misleading and confusing. I was interested in two things. One, which is easy, adding to the existing pool of nouns and adjectives. Two, which is open-ended and I haven't come up with any idea yet, is to create a new way to generate aliases which we can somehow say is a better way. Yes it's vague, but I think something might come up in discussion.

**Comment by gkeller2** [comment]
I was working on something last night. I'll send it to you today.

**Comment by gkeller2** [comment]
Just submitted a new change to the generator (a refactoring change).

**Comment by jrcamelo** [comment]
I did not change the generator itself, but i added some words and organized some of the entries there.
#24

**Comment by griffiwy** [comment]
Hey, is anyone working on this at the moment?
If not or they don't mind some help I would like to contribute.

**Comment by Mymaqn** [comment]
Hi.

I have compiled a list of 1339 adjectives that could be used as prefixes if it could be of any interest to add, if you are still working on this :)

**Comment by lauchokyip** [comment]
Hi I am new to this community. I am trying to use it and I have a question. Will it works on mobile phone? I am sending messages to the bot through my phone and it doesn't perform anything. Is it because I didn't set up?

**Comment by fear-the-reaper** [comment]
Hi I'm new here what exactly do you want to improve  how aliases are generated or more values in the list? Can you elaborate on what needs to be done. Thanks in advance

**Comment by Vyvy-vi** [comment]
Could I work on this?

**Comment by Vyvy-vi** [comment]
If I might ask, is the gender division logic necessary?
(this is supposed to be an anonymous messaging platform)
Moreover, in my humble opinion, the current alias generator looks quite **sexist**
(*I'm not pointing fingers or accusing, but you could take a look at the description lists and well last time I checked, `Detective`, `Sergeant`, `Connoisseur`, `Artist`, `Angel`, `General` etc.... and all the House Names from GOT were **gender-neutral**)

**Comment by mayukh18** [comment]
@Vyvy-vi You are correct. Thank you for pointing this out straight away. Although I kept this open without much scrutiny, just so people can have the experience of having a PR, it seems there are very bad examples creeping in. Honestly, I had faith people won't mess this up. 

I would clear up the bad clutter when I have some time. But since you are interested, I would be happy if you wanted to help with it.

**Comment by Vyvy-vi** [comment]
I did a rework of the generator. Slightly improved the randomisation and added a larger descriptor variety pool.
(Tho it's far from perfect as of now). Tried to keep gender classification limited to the first names

I plan on setting a name_length limit, to prevent the single names like 'Sailor', 'Baratheon' from appearing...
Here are a few samples:
![Screenshot 2020-12-31 at 10 33 55 PM](https://user-images.githubusercontent.com/62864373/103419295-7261a680-4bb8-11eb-83dd-e9bea5a48ac9.png)

**Comment by fear-the-reaper** [comment]
If you guy's don't mind can I aid in this venture. Sorry about the seixst
thing I didn't see it while I was working on it

**Comment by Vyvy-vi** [comment]
I have opened a draft PR with the changes
Once I'm done with improvements, it can be opened for review 👍

**Comment by mayukh18** [comment]
@fear-the-reaper You can make improvements once @Vyvy-vi's changes are merged. You can start now and rebase later too if needed.

**Comment by Chaitu17** [comment]
Hello @mayukh18 
Can I please know if my PR helps or not? This is my very first PR and very first contribution to open source. Please let me know if there are any mistakes that I've done. It helps me learn more about the open source community.

**Comment by DaanSelen** [comment]
What is the status of this issue?

**Comment by raghavnagar7** [comment]
To improve your alias generator, you can consider several enhancements to make it more sophisticated and versatile. Here are some ideas to take your alias generator to the next level:

1. **Randomness Control:** Implement various levels of randomness. For example, you can have a "randomness slider" that allows users to control how random or specific their alias should be.

2. **Customizable Templates:** Create templates for aliases. Users can select or customize these templates to create aliases with different structures. For example, you can have templates for "Hacker Alias," "Spy Alias," "Fantasy Character Alias," etc.

3. **Theme Selection:** Allow users to select themes for their aliases, such as sci-fi, fantasy, historical, or modern. Each theme could have predefined elements that fit the chosen setting.

4. **Character Traits:** Include options for users to define character traits for their aliases, such as introverted, extroverted, mysterious, etc. Generate aliases that reflect these traits.

5. **Name Meaning:** Provide users with the option to see the meaning or significance behind the alias generated for them. This can add depth to their chosen alias.

6. **User Input:** Allow users to input certain keywords or preferences, and incorporate them into the generated alias. For example, if a user wants their alias to include the word "dragon," the generator can ensure that it's part of the alias.

7. **Name Generator API Integration:** Integrate with external APIs or datasets to get access to a wider range of names, words, or phrases for generating aliases.

8. **Language Support:** Make your alias generator support multiple languages and character sets, allowing users from different backgrounds to create aliases.

9. **Save and Share:** Enable users to save their favorite aliases and share them with others. This could also foster a sense of community and creative sharing.

10. **AI Chatbot Integration:** Create a chatbot interface that can engage with users to understand their preferences and generate aliases based on a conversation. This adds a personalized touch.

11. **Feedback System:** Include a feedback system so users can rate and provide feedback on generated aliases. This can help improve the quality of future aliases.

12. **Profanity Filter:** Implement a profanity filter to ensure that generated aliases are appropriate and do not contain offensive language.

13. **Pop Culture References:** Incorporate elements from popular culture, such as movies, books, or games, into the alias generation process for added fun and creativity.

14. **Social Media Integration:** Allow users to link their generated aliases to social media profiles, websites, or avatars for quick and easy use.

15. **Mobile App:** Consider developing a mobile app for your alias generator to make it more accessible and user-friendly.

Remember that the key to making your alias generator better is to provide a wide range of options and customization while keeping the user experience simple and intuitive. Tailor your improvements to the specific needs and interests of your target audience.

**Comment by DaanSelen** [comment]
Maybe improve the alias generator with something like the introduction of namefake.com api?
